hnn_core/network.py

```diff
@@ -219,47 +305,408 @@ def __repr__(self):
+        distribution : str
+            Must be 'normal' (will be deprecated in a future release).
+        """
+        sim_end_time = self.cell_response.times[-1]
+        if tstop is None:
+            tstop = sim_end_time
+        if not self._legacy_mode:
+            _check_drive_parameter_values('bursty', tstart=tstart, tstop=tstop,
+                                          sim_end_time=sim_end_time,
+                                          sigma=tstart_std, location=location)
+            _check_drive_parameter_values('bursty', sigma=burst_std,
+                                          numspikes=numspikes,
+                                          spike_isi=spike_isi,
+                                          burst_rate=burst_rate)
+
+        drive = _NetworkDrive()
+        drive['type'] = 'bursty'
+        drive['cell_specific'] = False
+        drive['seedcore'] = seedcore
+
+        # XXX distribution='uniform' should be deprecated as it is used nowhere
+        # alternatively, create a new drive-type for it
+        drive['dynamics'] = dict(distribution=distribution, tstart=tstart,
+                                 tstart_std=tstart_std, tstop=tstop,
+                                 burst_rate=burst_rate, burst_std=burst_std,
+                                 numspikes=numspikes, spike_isi=spike_isi,
+                                 repeats=repeats)
+        drive['events'] = list()
+
+        self._attach_drive(name, drive, weights_ampa, weights_nmda, location,
+                           space_constant, synaptic_delays,
+                           cell_specific=False)
+
+    def _attach_drive(self, name, drive, weights_ampa, weights_nmda, location,
+                      space_constant, synaptic_delays, cell_specific=True):
+        """Attach a drive to network based on connectivity information
+
+        Parameters
+        ----------
+        name : str
+            Name of drive (must be unique)
+        drive : instance of _NetworkDrive
+            Collection of parameters defining the dynamics of the drive
+        weights_ampa : dict or None
+            Synaptic weights (in uS) of AMPA receptors on each targeted cell
+            type (dict keys). Cell types omitted from the dict are set to zero.
+        weights_nmda : dict or None
+            Synaptic weights (in uS) of NMDA receptors on each targeted cell
+            type (dict keys). Cell types omitted from the dict are set to zero.
+        location : str
+            Target location of synapses ('distal' or 'proximal')
+        space_constant : float
+            Describes lateral dispersion (from column origin) of synaptic
+            weights and delays within the simulated column
+        synaptic_delays : dict
+            Synaptic delay (in ms) at the column origin, dispersed laterally as
+            a function of the space_constant
+        cell_specific : bool
+            Whether each cell has unique connection parameters (default: True)
```

Comment:
It's really the cell type though right? As in this doesn't support single gid targeting... yet!

Reply:
Correct. I was on the fence on this one, opted for the short (but misleading) name. Changing to `cell_type_specific` for clarity.

Reply:
Oops, spoke too early... I actually _do_ mean cell-specific here! This is the old `p_unique` (vs. `p_common`) distinction. The flag controls whether each (real) cell in the network gets 'specific' input from a single artificial drive cell, or whether there is a one 'global' drive targeting all cells. In the latter case, the spike dynamics are identical across cells, though the connection strength and synaptic delay may differ (at the level of cell _types_, not individual cells, as you point out). Make sense?